contracts/pcv/maker/MakerPCVSwapper.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {IDSSPSM} from "./IDSSPSM.sol";
+import {IPCVSwapper} from "../IPCVSwapper.sol";
+import {CoreRefV2} from "../../refs/CoreRefV2.sol";
+
+/// @notice This contracts allows swaps between DAI and USDC through Maker's DAI PSM.
+/// @author eswak
+contract MakerPCVSwapper is IPCVSwapper, CoreRefV2 {
+    /// @notice reference to the Maker DAI-USDC PSM
+    address public constant MAKER_PSM =
+        0x89B78CfA322F6C5dE0aBcEecab66Aee45393cC5A;
+
+    /// @notice reference to the contract used to sell USDC for DAI
+    address public constant MAKER_GEM_JOIN =
+        0x0A59649758aa4d66E25f08Dd01271e891fe52199;
+
+    /// @notice reference to the DAI contract used.
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+
+    /// @notice reference to the USDC contract used.
+    address public constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+
+    /// @notice scaling factor for USDC
+    uint256 public constant USDC_SCALING_FACTOR = 1e12;
+
+    constructor(address _core) CoreRefV2(_core) {}
+
+    // ----------- IPCVSwapper View API ---------------------
+
+    function canSwap(
+        address assetIn,
+        address assetOut
+    ) external view returns (bool) {
+        return
+            !paused() &&
+            ((assetIn == DAI && assetOut == USDC) ||
+                (assetIn == USDC && assetOut == DAI));
+    }
+
+    // ----------- IPCVSwapper State-changing API -----------
+
+    function swap(
+        address assetIn,
+        address assetOut,
+        address destination
+    ) external whenNotPaused returns (uint256) {
```

Comment:
do we want to restrict who can call this function, that way if funds are sent here and swap isn't called funds aren't lost?

Reply:
another idea is do we want this function to restrict destination addresses to only currently active PCV deposits in the PCV Oracle? that way this cannot possibly send funds out of the system even if funds get parked in this particular contract.

Reply:
I'd like to keep these contracts stateless and unprotected, they are already interacted with through protocol contracts that can check destination and make sure they atomically transfer/swap

the swappers are not listed as PCV deposits and are not listed as safe address so they cannot receive funds other than through the PCVRouter, and the PCVRouter already performs checks on destination and is sure to atomically send/swap